www/include/views/graphs/graph-periods.php

```diff
@@ -52,52 +52,53 @@
 $tpl = new Smarty();
 $tpl = initSmartyTpl($path, $tpl);
 
-function getGetPostValue($str)
-{
-    $value = null;
-    if (isset($_GET[$str]) && $_GET[$str]) {
-        $value = $_GET[$str];
-    }
-    if (isset($_POST[$str]) && $_POST[$str]) {
-        $value = $_POST[$str];
-    }
-    return urldecode($value);
+if (isset($_GET['chartId'])) {
+    $chartId = filter_var($_GET['chartId'], FILTER_SANITIZE_STRING);
```

Comment:
You should also check that $chartId is not empty after filter_var.  FILTER_SANITIZE_STRING can't return false but could return an empty string.
```suggestion
    $chartId = filter_var($_GET['chartId'], FILTER_SANITIZE_STRING);
    if (empty($chartId)) {
        throw new \InvalidArgumentException('Invalid chartId format');
    }
```

But I don't think its mandatory in this case, as you do a preg_match later.

Reply:
it is required, as $chartId is forwarded to the html here :
https://github.com/centreon/centreon/pull/9735/files#diff-b83ceac4abbf0d188b342b930198fe3848959d1b701e1436a3b927c3cacfee62R101